include/deal.II/base/symmetric_tensor.h

```diff
@@ -835,11 +853,8 @@ class SymmetricTensor
   template <int dim2, typename Number2>
   friend SymmetricTensor<4,dim2,Number2> identity_tensor ();
 
-  template <int dim2, typename Number2>
-  friend SymmetricTensor<2,dim2,Number2> invert (const SymmetricTensor<2,dim2,Number2> &);
-
-  template <int dim2, typename Number2>
-  friend SymmetricTensor<4,dim2,Number2> invert (const SymmetricTensor<4,dim2,Number2> &);
+  template <int,int,typename>
+  friend struct internal::SymmetricTensor::Inverse;
```

Comment:
So do you mean that you want me to declare the following friends:
```c++
  friend struct internal::SymmetricTensor::Inverse<2,dim,Number>;
  friend struct internal::SymmetricTensor::Inverse<4,dim,Number>;
```
? Its either the generalisation or what I've written above because one cannot declare partial specialisations as friends:
```c++
  template <int dim2, typename Number2>
  friend struct internal::SymmetricTensor::Inverse<2,dim2,Number2>;
  template <int dim2, typename Number2>
  friend struct internal::SymmetricTensor::Inverse<4,dim2,Number2>;
```
leads to
```sh
<>/dealii/include/deal.II/base/symmetric_tensor.h:857:10: error: partial specialization cannot be declared as a friend
  friend struct internal::SymmetricTensor::Inverse<2,dim2,Number2>;
         ^                                        ~~~~~~~~~~~~~~~~
<>/dealii/include/deal.II/base/symmetric_tensor.h:860:10: error: partial specialization cannot be declared as a friend
  friend struct internal::SymmetricTensor::Inverse<4,dim2,Number2>;
         ^                                        ~~~~~~~~~~~~~~~~
```
I've now modified this. I'm not sure if there are any larger implications stemming from this change (i.e. what are the other friend _functions_ templated?), but it seems to compile successfully and the tensor-related tests ran fine on my machine.